stdlib/public/runtime/Metadata.cpp

```diff
@@ -56,19 +62,45 @@
 using namespace swift;
 using namespace metadataimpl;
 
+// allocate memory up to a nearby page boundary
+static void *swift_allocPage(size_t size) {
+#if defined(_MSC_VER)
+  auto mem = VirtualAlloc(
+      nullptr, size, MEM_TOP_DOWN | MEM_RESERVE | MEM_COMMIT, PAGE_READWRITE);
+#else
+  auto mem = mmap(nullptr, size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE,
+                  VM_TAG_FOR_SWIFT_METADATA, 0);
+  if (mem == MAP_FAILED)
+    mem = nullptr;
+#endif
+  return mem;
+}
+
+// free memory allocated by swift_allocPage()
+// On success, swift_freePage() returns 0, on failure -1
+static int swift_freePage(void *addr, size_t size) {
```

Comment:
Seems like we are not checking the return value of this function.  Could you change it to return void and trap if the system call failed?